src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -292,10 +297,86 @@ bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
                     return false;
                 }
 
-                // Check truncated immediates
+                if (opcode is Instruction.RJUMP or Instruction.RJUMPI)
+                {
+                    if (pos + IMMEDIATE_16BIT_BYTE_COUNT > code.Length)
+                    {
+                        if (Logger.IsTrace)
+                        {
+                            Logger.Trace($"EIP-4200 : Static Relative Jumpv Argument underflow");
+                        }
+                        return false;
+                    }
+
+                    var offset = code.Slice(pos, IMMEDIATE_16BIT_BYTE_COUNT).ReadEthInt16();
+                    immediates.Add(new Range(pos, pos + 1));
+                    var rjumpdest = offset + IMMEDIATE_16BIT_BYTE_COUNT + pos;
+                    rjumpdests.Add(rjumpdest);
+                    if (rjumpdest < 0 || rjumpdest >= code.Length)
+                    {
+                        if (Logger.IsTrace)
+                        {
+                            Logger.Trace($"EIP-4200 : Static Relative Jump Destination outside of Code bounds");
+                        }
+                        return false;
+                    }
+                    pos += IMMEDIATE_16BIT_BYTE_COUNT;
+                }
+
+                if (opcode is Instruction.RJUMPV)
+                {
+                    if (pos + IMMEDIATE_16BIT_BYTE_COUNT > code.Length)
```

Comment:
``pos + IMMEDIATE_16BIT_BYTE_COUNT > code.Length`` is exaggerated a bit, because ``JUMPV_COUNT_BYTE_COUNT == 1`` and for JUMPV Immediate value size is dynamic based on the Count byte, I renamed  ``JUMPV_COUNT_BYTE_COUNT `` to ``ONE_BYTE_LENGTH``

Reply:
Ah I find this part confusing: 

https://github.com/NethermindEth/nethermind/blob/b9bbc6195f5cc968742a27ea0205658fa5453190/src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs#L291-L292

`pos` is not the current position it is the next position. Not sure if this is great for readability.

Reply:
Maybe rename `pos` to 'nextByteOffset' or something?